src/runtime/Runtime.cs

```diff
@@ -351,6 +357,7 @@ static bool TryCollectingGarbage(int runs, bool forceBreakLoops)
                 {
                     NullGCHandles(CLRObject.reflectedObjects);
                     CLRObject.reflectedObjects.Clear();
+                    CLRObject.creationBlocked = true;
```

Comment:
I think this needs to be moved to the callsite instead. E.g. immediately after `TryCollectingGarbage(forceBreakLoops: true)`.

Otherwise if you have a Python object that needs to temporarily create a .NET object in its destructor (for instance write log summary to a file), its destructor will fail if it happens to be freed on the second iteration of loop breaking.